v2/internal/identity/credential_provider.go

```diff
@@ -334,6 +335,45 @@ func (c *credentialProvider) newCredentialFromSecret(secret *v1.Secret) (*Creden
 		}, nil
 	}
 
+	// This authentication type is similar to user assigned managed identity authentication combined with client certificate
+	// authentication. As a 1st party Microsoft application, one has access to pull a user assigned managed identity's backing
+	// certificate information from the MSI data plane. Using this data, a user can authenticate to Azure Cloud.
+	if userAssignedCredentialsPath, hasUserAssignedCredentials := secret.Data[config.AzureUserAssignedIdentityCredentials]; hasUserAssignedCredentials {
+		// Default to AzurePublic
+		options := azcore.ClientOptions{
```

Comment:
You shouldn't need to do the defaulting of AuthorityHost here, as it's done for you automatically now and is accessible via `c.cloud` (this was just added by my earlier PR, it wasn't here previously)

Reply:
c.cloud is set based on the global configmaps cloud configuration. Our intent here was that you can't cross clouds with a single instance of ASO, it always uses the globals cloud configuration for per-namespace or per-resource secrets.

Do you have a requirement that you be able to cross clouds w/ the per-resource or per-namespace credential?

Reply:
That was an oversight on my part not including that here. We do not have any requirement to cross clouds.